sockeye/quantization.py

```diff
@@ -0,0 +1,170 @@
+                 weight_initializer=None, bias_initializer='zeros',
+                 in_units=0, **kwargs):
+        super(QuantizableDense, self).__init__(**kwargs)
+        self._flatten = flatten
+        self._dtype = dtype
+        with self.name_scope():
+            self._units = units
+            self._in_units = in_units
+            if dtype == C.DTYPE_INT8:
+                self.scaling = self.params.get('scaling', shape=(1,),
+                                               init='zeros', dtype=C.DTYPE_FP32,
+                                               allow_deferred_init=True)
+                weight_initializer = 'zeros' # Most initializers don't work for int8, but this is for inference anyway.
+
+            self.weight = self.params.get('weight', shape=(units, in_units),
+                                          init=weight_initializer, dtype=dtype,
+                                          allow_deferred_init=True)
+
+            if use_bias:
+                self.bias = self.params.get('bias', shape=(units,),
+                                            init=bias_initializer, dtype = C.DTYPE_FP32,
+                                            allow_deferred_init=True)
+            else:
+                self.bias = None
+            if activation is not None:
+                self.act = Activation(activation, prefix=activation+'_')
+            else:
+                self.act = None
+
+    def cast(self, dtype):
+        if self._dtype != C.DTYPE_INT8:
+            self._dtype = dtype
+            super(QuantizableDense, self).cast(dtype)
+        else:
+            #No casting an already quantized matrix.
+            logger.warning("Ignoring casting on int8 matrix")
+
+    def hybrid_forward(self, F, x, weight, scaling = None, bias=None):
+        if self._dtype == C.DTYPE_INT8:
+            act = F.contrib.intgemm_fully_connected(x, weight, scaling, bias, no_bias=bias is None, num_hidden=self._units,
+                                                    flatten=self._flatten, name='fwd')
+        else:
+            #Newer MXNet allows a numpy array.
+            #fc = F.npx.fully_connected if is_np_array() else F.FullyConnected
+            act = F.FullyConnected(x, weight, bias, no_bias=bias is None, num_hidden=self._units,
+                     flatten=self._flatten, name='fwd')
+        if self.act is not None:
+            act = self.act(act)
+        return act
+
+    def __repr__(self):
+        s = '{name}({layout}, {act})'
+        shape = self.weight.shape
+        return s.format(name=self.__class__.__name__,
+                        act=self.act if self.act else 'linear',
+                        layout='{0} -> {1}'.format(shape[1] if shape[1] else None, shape[0]))
+
+
+#Convert weights from float32 MXNet format (B^T in float32) to disk format (B^T in int8 format).
+#params is expected to be model.collect_params() from a float32 model
+def convert_weights_disk_format(params):
```

Comment:
you could add a type annotation: 
```suggestion
def convert_weights_disk_format(params: mx.gluon.parameter.ParameterDict):
```